bin/autodetect/Main.cc

```diff
@@ -275,17 +282,16 @@ int main(int argc, char** argv) {
     // Chain the categorizer's output to the anomaly detector's input
     ml::api::COutputChainer outputChainer(job);
 
-    ml::api::CJsonOutputWriter fieldDataCategorizerOutputWriter(jobId, wrappedOutputStream);
```

Comment:
This is making the categorizer consistent with the anomaly detector.  In the beginning, both shared the same `CJsonOutputWriter`.  When forecasting was added the anomaly detector was given a separate `CJsonOutputWriter` that it manages internally.  So the global `CJsonOutputWriter` was then for categorization only (and maybe nothing if categorization was not in use), leading to asymmetry.  Moving it into the categorizer means that it can be finalised when the categorizer is finalised, which in turn means the "Unfortunately we cannot rely on destruction to finalise" comment and associated code smell can be removed.